core/plugins/org.polarsys.capella.core.data.fa.validation/src/org/polarsys/capella/core/data/fa/validation/function/MDCHK_RootFunction_FunctionRealization.java

```diff
@@ -41,16 +40,16 @@ public IStatus validate(IValidationContext ctx) {
     if (eType == EMFEventType.NULL) {
       if (eObj instanceof AbstractFunction) {
         AbstractFunction function = (AbstractFunction) eObj;
-        String currentFunctionName = CapellaElementExt.getValidationRuleMessagePrefix(function);
+        String currentFunctionName = "Root \"" + function.getName() + "\" (" + function.eClass().getName() + ") ";

```

Comment:
please use EObjectLabelProviderHelper.getMetaclassLabel. it will retrieve 'System Function'.

Reply:
not sure you need to add Root keyword.

Reply:
Shyam asked to add the keyword "Root"

Reply:
currentFunctionName = NLS.bind("Root {0} ({1])", function.getName(), EOLPH.getMetaclassLabel(function))